### PR TITLE
171 signup route

### PR DESCRIPTION
**Pull request by Zimbardoex**
#### Solves:
* #171

### CODE REVIEW

File: express-back-end/src/api/auth/controllers/users.controller.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          value: ["email"],
+          message: "Email already taken",
+        }]
+      });
+      return;
+    }
+
+    // Hash the password
+    const salt = await bcrypt.genSalt(10);
+    const hashedPassword = await bcrypt.hash(UserData.password,salt);
```
Comment (horrjd1) [inline express-back-end/src/api/auth/controllers/users.controller.js]: This should probably be extracted to a function

File: express-back-end/src/api/auth/controllers/users.controller.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        {
+          _id: result._id,
+          name: result.name,
+          email: result.email,
+        },
+        config.secret,
+        {
+          expiresIn: '4h',
+        },
+      );
```
Comment (horrjd1) [inline express-back-end/src/api/auth/controllers/users.controller.js]: Could probably extract this to a function

File: express-back-end/src/api/auth/controllers/users.controller.js (hunk 1 above)
Comment (NICHTJ3) [reply express-back-end/src/api/auth/controllers/users.controller.js]: This has been done on the login branch

File: express-back-end/src/api/auth/controllers/users.controller.js (hunk 2 above)
Comment (NICHTJ3) [reply express-back-end/src/api/auth/controllers/users.controller.js]: This has been done on the login branch

**Comment by NICHTJ3** [comment]
We will add the tests in a later pr

**Pull request merged**